File: frontend/lib/models/task_log.dart
//db table
//  TASK_LOG {
//         Long id PK
//         Long task_id FK
//         String action "CREATED, ASSIGNED, SHOPPING_STARTED, DELIVERED, COMPLETED"
//         Timestamp timestamp
//     }

enum Action {
  created,
  assigned,
  shoppingStarted,
  delivered,
  completed,
  cancelled;

  static Action fromString(String action) {
    return Action.values.firstWhere((e) => e.name == action);
  }
}

class TaskLog {
  final String id;
  final String taskId;
  final Action action;
  final DateTime timestamp;

  TaskLog({
    required this.id,
    required this.taskId,
    required this.action,
    required this.timestamp,
  });

  factory TaskLog.fromJson(Map<String, dynamic> json) {
    return TaskLog(
      id: json['id'],
      taskId: json['task_id'],
      action: Action.fromString(json['action']),
      timestamp: json['timestamp'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'task_id': taskId,
      'action': action.name,
      'timestamp': timestamp,
    };
  }
}

File: backend/src/main/java/com/gencel/backend/entity/TaskLog.java
package com.gencel.backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "task_logs")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class TaskLog {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "task_id", nullable = false)
    private Task task;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TaskLogAction action;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id") // Optional: User who performed the action
    private User user;

    @Column(columnDefinition = "TEXT")
    private String details;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDateTime timestamp;

    public enum TaskLogAction {
        CREATED, ASSIGNED, SHOPPING_STARTED, DELIVERED, COMPLETED, CANCELLED
    }
}
